src/transformers/modeling_tf_auto.py

```diff
@@ -654,11 +738,13 @@ def from_pretrained(cls, pretrained_model_name_or_path, *model_args, **kwargs):
 
         """
         config = kwargs.pop("config", None)
+
         if not isinstance(config, PretrainedConfig):
             config = AutoConfig.from_pretrained(pretrained_model_name_or_path, **kwargs)
 
         for config_class, model_class in TF_MODEL_WITH_LM_HEAD_MAPPING.items():
-            if isinstance(config, config_class):
+            # Not using isinstance() here to do not take into account inheritance
+            if config_class == type(config):
```

Comment:
in pytorch the different configs are sorted so that you never get a child class before its parent (precisely to prevent this), but this is reasonable solution too